File: frontend/test_api.js
const login = async () => {
    const res = await fetch('http://127.0.0.1:8000/api/v1/auth/login', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
        },
        body: new URLSearchParams({
            'username': 'kflorimon@gmail.com',
            'password': '123'
        })
    });
    const data = await res.json();
    return data.access_token;
};

const run = async () => {
    try {
        const token = await login();
        console.log("Logged in");
        
        const headers = {
            'Authorization': `Bearer ${token}`,
            'X-Centro-Id': '1'
        };
        
        const usersRes = await fetch('http://127.0.0.1:8000/api/v1/usuarios/', { headers });
        const users = await usersRes.json();
        
        const docente = users.find(u => u.nombre_completo.includes('Kelvin') && u.roles.includes('DOCENTE'));
        console.log("Docente ID:", docente?.docente_id);
        
        if (docente) {
            const asigsRes = await fetch(`http://127.0.0.1:8000/api/v1/asignaciones/docente/${docente.docente_id}`, { headers });
            const asigs = await asigsRes.json();
            console.log("Asignaciones docente:");
            console.log(JSON.stringify(asigs, null, 2));
        }
        
        console.log("----");
        
        const seccRes = await fetch('http://127.0.0.1:8000/api/v1/secciones/', { headers });
        const secciones = await seccRes.json();
        
        for (const s of secciones) {
            const asigSeccRes = await fetch(`http://127.0.0.1:8000/api/v1/asignaciones/seccion/${s.id}`, { headers });
            const asigSecc = await asigSeccRes.json();
            if (asigSecc.length > 0) {
                console.log(`Sección ${s.nombre} (ID: ${s.id}):`);
                console.log(JSON.stringify(asigSecc, null, 2));
            }
        }
        
    } catch(e) {
        console.error(e);
    }
};

run();

File: frontend/test_api.py
import requests
import os
from dotenv import load_dotenv

load_dotenv(dotenv_path='../.env')

API_URL = 'http://localhost:8000/api/v1'

# 1. First get a valid token (superuser login)
try:
    print("Iniciando sesión...")
    res = requests.post(f"{API_URL}/auth/login", data={"username": "kflorimon@gmail.com", "password": "123"})
    res.raise_for_status()
    token = res.json()["access_token"]
    headers = {"Authorization": f"Bearer {token}", "X-Centro-Id": "1"}
    
    # 2. Let's find the Docente ID for Kelvin Florimon
    print("\n--- Usuarios y Docentes ---")
    res = requests.get(f"{API_URL}/usuarios/", headers=headers)
    usuarios = res.json()
    docente_user = next((u for u in usuarios if "Kelvin Florimon" in u.get("nombre_completo", "") and "DOCENTE" in u.get("roles", [])), None)
    
    if not docente_user:
        print("No se encontró al docente.")
    else:
        print(f"Docente encontrado: {docente_user}")
        docente_id = docente_user["docente_id"]
        
        # 3. Check assignments for this docente
        print(f"\n--- Asignaciones del Docente (ID: {docente_id}) ---")
        res = requests.get(f"{API_URL}/asignaciones/docente/{docente_id}", headers=headers)
        if res.status_code == 200:
            print("Status 200")
            print(res.json())
        else:
            print(f"Error: {res.status_code} - {res.text}")
            
    # 4. Check assignments by section
    print("\n--- Secciones y Asignaciones ---")
    res = requests.get(f"{API_URL}/secciones/", headers=headers)
    secciones = res.json()
    for s in secciones:
        print(f"Sección {s['nombre']} (ID: {s['id']}, Grado ID: {s['grado_id']})")
        res_asig = requests.get(f"{API_URL}/asignaciones/seccion/{s['id']}", headers=headers)
        if res_asig.status_code == 200:
            asigs = res_asig.json()
            if asigs:
                for a in asigs:
                    print(f"  - Asignatura {a['asignatura_nombre']} (ID: {a.get('asignatura_id')}) -> Docente: {a['docente_nombre']} (ID: {a.get('docente_id')})")
            else:
                print("  - Sin asignaciones")
                
except Exception as e:
    print(f"Error: {e}")
